scraper.py

```diff
@@ -79,6 +79,15 @@ def is_valid(url):
         if parsed.scheme not in set(["http", "https"]):
             return False
 
+        #check if the path is a calendar because they are traps
+
+        if re.match(r'\/calendar\/.+ | \/events\/.+', parsed.path.lower()):
+            return False
+
+        #check if there are duplicate folder paths
+        if re.match(r'.*?\/(.+?)\/.?\1.* | .*?\/(.+?)\/.?\2.*'. parsed.path.lower()) or re.match(r'.*\..+\/', pared.path.lower()):
+            return False
+

```

Comment:
Line 88. Typo. I think: pared.path.lower(), you meant parsed right?

I think that you should break this code out into a separate function isTrap(parsed) . 
1. For code readability
2. I doubt this will be the last of the traps we see.

Reply:
okay will fix that.